crates/sui/src/sui_commands.rs

```diff
@@ -1381,6 +1413,120 @@ fn read_line() -> Result<String, anyhow::Error> {
     Ok(s.trim_end().to_string())
 }
 
+async fn get_chain_id_and_client(
+    client_config: SuiEnvConfig,
+    command_err_string: &str,
+) -> anyhow::Result<(Option<String>, Option<SuiClient>)> {
+    let config = client_config
+        .config
+        .unwrap_or(sui_config_dir()?.join(SUI_CLIENT_CONFIG));
+    prompt_if_no_config(&config, false).await?;
+    let mut context = WalletContext::new(&config)?;
+
+    if let Some(env_override) = client_config.env {
+        context = context.with_env_override(env_override);
+    }
+
+    let Ok(client) = context.get_client().await else {
+        bail!(
+            "`{command_err_string}` requires a connection to the network. \
+             Current active network is {} but failed to connect to it.",
+            context.config.active_env.as_ref().unwrap()
+        );
+    };
+
+    if let Err(e) = client.check_api_version() {
+        eprintln!("{}", format!("[warning] {e}").yellow().bold());
+    }
+
+    Ok((
+        client.read_api().get_chain_identifier().await.ok(),
+        Some(client),
+    ))
+}
+
+/// Try to resolve an ObjectID to a MovePackage
+async fn resolve_package(reader: &ReadApi, package_id: ObjectID) -> anyhow::Result<MovePackage> {
+    let object = reader
+        .get_object_with_options(package_id, SuiObjectDataOptions::bcs_lossless())
+        .await?
+        .into_object()?;
+
+    let Some(SuiRawData::Package(package)) = object.bcs else {
+        bail!("Object {} is not a package.", package_id);
+    };
+
+    Ok(MovePackage::new(
+        package.id,
+        package.version,
+        package.module_map,
+        // This package came from on-chain and the tool runs locally, so don't worry about
+        // trying to enforce the package size limit.
+        u64::MAX,
+        package.type_origin_table,
+        package.linkage_table,
+    )?)
+}
+
+async fn download_package_and_deps_under(
+    read_api: &ReadApi,
+    path: &Path,
+    package_id: ObjectID,
+) -> anyhow::Result<PackageSummaryMetadata> {
```

Comment:
nit: move this to use anyhow::Result;
```suggestion
) -> Result<PackageSummaryMetadata> {
```